nomad/rpc.go

```diff
@@ -793,7 +793,8 @@ func (r *rpcHandler) setQueryMeta(m *structs.QueryMeta) {
 		m.KnownLeader = true
 	} else {
 		m.LastContact = time.Since(r.raft.LastContact())
-		m.KnownLeader = (r.raft.Leader() != "")
+		knownLeader, _ := r.raft.LeaderWithID()
+		m.KnownLeader = (knownLeader != "")
```

Comment:
```suggestion
		leaderAddr, _ := r.raft.LeaderWithID()
		m.KnownLeader = (leaderAddr != "")
```

I think this makes it a little more clear what's being tested against the empty string?